cmd/gencerts_test.go

```diff
@@ -0,0 +1,122 @@
+package cmd_test
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/OpenBazaar/openbazaar-go/cmd"
+	"github.com/OpenBazaar/openbazaar-go/schema"
+)
+
+func buildCertDirectory() (*string, func(), error) {
+	ctx := schema.SchemaContext{
+		DataPath:        schema.GenerateTempPath(),
+		TestModeEnabled: true,
+	}
+	certSchema := schema.MustNewCustomSchemaManager(ctx)
+	if err := certSchema.BuildSchemaDirectories(); err != nil {
+		return nil, nil, err
+	}
+	return &ctx.DataPath, certSchema.DestroySchemaDirectories, nil
+}
+
+func TestGenCertsGenericDefaults(t *testing.T) {
+	var dataPath, Destroy, schema_err = buildCertDirectory()
+	dataDir := string([]byte(*dataPath))
+	if schema_err != nil {
+		Destroy()
+		t.Fatal(schema_err)
+	}
+
+	config := cmd.GenerateCertificates{
+		DataDir:  dataDir,
+		Testnet:  true,
+		Host:     "127.0.0.1",
+		ValidFor: 1, // 1ns ... 1*1e9 == 1s
+	}
+
+	args := []string{"args?"}
+
+	if err := config.Execute(args); err != nil {
+		Destroy()
+		t.Fatal(err)
+	}
+
+	sslPath := path.Join(dataDir, "ssl")
+	if fileInfo, err := os.Stat(sslPath); err != nil {
+		Destroy()
+		t.Fatal(err)
+	} else {
+		if fileInfo.Mode().Perm() != 0755 {
+			Destroy()
+			t.Fatal("ssl directory does not have 0755 permissions")
+		}
+		if !fileInfo.IsDir() {
+			Destroy()
+			t.Fatalf("Expecting a directory: %s", dataDir)
+		}
+	}
```

Comment:
Remember that `Fatal` stops the test. Because the if would stop the test as part of the call to `Fatal` we don't need the else here as it's implied that those two checks will be performed if `err != nil` is false in the check above. I realize the `else` extends the scope of the `fileInfo` and `err` variables, but it's preferred to have logic that is "not nested". Rewriting that would look like:
```
fileInfo, err := os.Stat(sslPath)
if err != nil {
	t.Fatal(err)
}
if fileInfo.Mode().Perm() != 0755 {
	t.Fatal("ssl directory does not have 0755 permissions")
}
if !fileInfo.IsDir() {
	t.Fatalf("Expecting a directory: %s", dataDir)
}
```

(Note: the Destroy() is gone because we used `defer Destroy()` as described above.)